tests/click_types.py

```diff
@@ -29,7 +28,22 @@
 from globus_cli.types import JsonValue
 
 
-def _param2types(param_obj: click.Parameter) -> tuple[type, ...]:
+def _paramtypes_from_param_type(param_obj: click.Parameter) -> tuple[type, ...]:
+    """
+    Given a Parameter instance, read the 'type' attribute and deduce the tuple of
+    possible types which it describes
+
+    Supports both `click` native types and `globus_cli` custom types
+
+    For example:
+        IntParamType -> (int,)
+        IntRange -> (int,)
+        StringOrNull -> (str, None)
+
+    Returning types as a tuple rather than building a union when there is more than one
+    ensures that we build "flat" unions of the form `Union[x, y, z]` and avoid nexted
```

Comment:
```suggestion
    ensures that we build "flat" unions of the form `Union[x, y, z]` and avoid nested
```

Reply:
Thanks! I just amended to include this. 🤫